src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,271 @@
+                self.doc_deprecated.clone()
+            }
+        }
+    }
+}
+
+impl_to_stripper_type!(Alias, Struct);
+impl_to_stripper_type!(Constant, Const);
+impl_to_stripper_type!(Member, Variant);
+impl_to_stripper_type!(Enumeration, Enum);
+impl_to_stripper_type!(Bitfield, Type);
+impl_to_stripper_type!(Record, Type);
+impl_to_stripper_type!(Field, Variant);
+impl_to_stripper_type!(Union, Struct);
+impl_to_stripper_type!(Function, Fn);
+impl_to_stripper_type!(Interface, Trait);
+impl_to_stripper_type!(Class, Struct);
+impl_to_stripper_type!(Namespace, Mod);
+impl_to_stripper_type!(Parameter, Variant);
+
+pub fn generate(env: &Env) {
+    let path =  env.config.target_path.join("doc.cmts");
+    println!("Generating documentation {:?}", path);
+    save_to_file(&path, env.config.make_backup,
+        |w| generate_doc(w, &env.library));
+}
+
+fn generate_doc(mut w: &mut Write, lib: &Library) -> Result<()> {
+    let namespace = lib.namespace(MAIN);
+
+    /*if let Some(ref namespace_doc) = namespace.doc {
+        for line in namespace_doc.split("\n") {
+            output.push_str(&format!("{}{}\n", FILE_COMMENT, line));
+        }
+    }
+    try!(create_sub_docs(&mut output, &namespace.constants, None));*/
+    for ty in namespace.types.iter().filter_map(|t| t.as_ref()) {
+        try!(handle_type(&mut w, &ty, None, lib));
+    }
+    Ok(())
+}
+
+fn handle_type(w: &mut Write, ty: &LType, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    match *ty {
+        LType::Alias(ref a) => {
+            try!(writeln!(w, "{}src/auto/{}.rs", FILE, a.name.to_lowercase()));
+            create_sub_doc(w, a, parent)
+        },
+        LType::Enumeration(ref e) => create_enum_doc(w, &e, parent),
+        LType::Function(ref f) => create_fn_doc(w, &f, parent),
+        LType::Interface(ref i) => create_trait_doc(w, &i, parent),
+        LType::Class(ref c) => create_class_doc(w, &c, parent, lib),
+        _ => Ok(()),
+    }
+}
+
+fn create_class_doc(w: &mut Write, class: &Class, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    let tabs : String = primitives::tabs(compute_indent(&parent));
+    let mut ty = class.convert();
+    ty.parent = parent;
+
+    try!(writeln!(w, "{}src/auto/{}.rs", FILE, class.name.to_lowercase()));
+    if class.doc().is_some() || class.doc_deprecated().is_some() {
+        try!(writeln!(w, "{}{}", MOD_COMMENT, ty));
+    }
+
+    if let Some(ref class_doc) = class.doc {
+        try!(write_lines(w, &class_doc, &tabs));
+    }
+    if let Some(ref class_doc) = class.doc_deprecated {
+        try!(write_deprecated(w, &tabs));
+        try!(write_lines(w, &class_doc, &tabs));
+    }
+
+    for function in class.functions.iter() {
+        try!(create_fn_doc(w, &function, Some(Box::new(ty.clone()))));
+    }
+    for child_id in class.children.iter() {
+        if let &LType::Class(ref child) = lib.type_(*child_id) {
+            try!(create_class_doc(w, &child, None, lib));
```

Comment:
Why is `create_class_doc` recursive? I've now confirmed my suspicion that it generates redundant copies of documentation.